peekingduck/pipeline/nodes/dabble/camera_calibration.py

```diff
@@ -435,99 +301,301 @@ def _calculate_coeffs(self, img_shape: tuple) -> None:
             error /= len(projected_image_points)
             mean_error += error
 
-        self.logger.info(f"total eror: {mean_error / len(self.object_points)}")
-
-    def run(self, inputs: Dict[str, Any]) -> Dict[str, Any]:  # type: ignore
-        """This node calculates the camera distortion coefficients for undistortion.
+        self.logger.info(f"Total error: {mean_error / len(self.object_points)}")
+
+    def _draw_text_and_corners(
+        self,
+        img: np.ndarray,
+        gray_img: np.ndarray,
+        corners: np.ndarray,
+        text_pos: tuple,
+    ) -> None:
+        """Draws text and corners on image"""
+
+        # improve corner accuracy
+        corners_accurate = cv2.cornerSubPix(
+            image=gray_img,
+            corners=corners,
+            winSize=(11, 11),
+            zeroZone=(-1, -1),
+            criteria=TERMINATION_CRITERIA,
+        )
 
-        Args:
-            inputs (dict): Inputs dictionary with the key `img`.
+        # draw corners and message on the image
+        cv2.drawChessboardCorners(
+            image=img,
+            patternSize=self.num_corners,
+            corners=corners_accurate,
+            patternWasFound=True,
+        )
 
-        Returns:
-            outputs (dict): Outputs dictionary with the key `img`.
-        """
+        if self.num_detections != NUM_PICTURES:
+            text_to_draw = DETECTION_SUCCESS
+        else:
+            text_to_draw = DETECTION_COMPLETE
 
-        img = inputs["img"]
-        img = cv2.flip(img, 1)
-        gray_img = cv2.cvtColor(img, cv2.COLOR_BGR2GRAY)
+        _draw_text(
+            img, text_to_draw, text_pos, self.display_scales["normal_font_scale"]
+        )
 
-        detect_corners_success = False
+        # display the image and wait for user to press a key
+        cv2.imshow("PeekingDuck", img)
+        cv2.waitKey(0)
 
-        height, width = img.shape[:2]
+    def _draw_text_and_countdown(
+        self, img: np.ndarray, text_to_draw: List[str], text_pos: tuple
+    ) -> None:
+        """Draws text and countdown on image"""
 
-        start_point, end_point, text_pos = _get_box_info(
-            self.num_detections, width, height
+        _draw_text(
+            img, text_to_draw, text_pos, self.display_scales["normal_font_scale"]
         )
 
-        img_notext = _draw_box(img, start_point, end_point)
-        img = _draw_text(
-            img_notext,
-            f"{DEFAULT_TEXT} ({self.num_detections+1}/{self.num_pictures})",
-            text_pos,
+        time_to_next_detection = math.ceil(5 - time.time() + self.last_detection)
+        if time_to_next_detection > 0:
+            _draw_countdown(
+                img, time_to_next_detection, self.display_scales["countdown_font_scale"]
+            )
+
+
+def _get_optimal_font_scale(text: str, width: int) -> float:
+    """Calculate optimal font scale given text and width"""
+    for scale in range(250, 0, -1):
```

Comment:
Good that this loop only needs to run once instead of running for every frame now. However, for the following reasons I'm now wondering if we should remove this totally:

We should cater for a wide range of [camera resolutions](https://optiviewusa.com/cctv-video-resolutions/). Let's consider the following 4 common types:
1) 4D UltraHD - 3840 x 2160
2) 1080p HD - 1920 x 1080
3) 720p HD - 1280 x 720
4) VGA - 640 x 480

You can simulate any of these using the `resize` parameter for `input.visual` (make sure you set `do_resizing` to `True` as well).

I noticed that the current implementation does not increase the thickness once the width goes beyond 1280 pixels, which creates a problem for the higher res cameras. To fix this, we would have to increase the upper bound of the loop from `250` to about `1100`. While this could run pretty fast on a modern computer, let's take a step back - could we perhaps be over-complicating the problem by constantly checking `cv2.getTextSize()` with a feedback loop?

Can we use a simple approach of ratios without a feedback loop as shown here? I used 1280 width as a baseline and used this - it works okay for 3 out of the 4 resolutions above. For VGA, the `MAX_LEN_TEXT` slightly overlaps with the box - possibly by considering the spacing required, this can work too.
```
            self.display_scales["normal_font_scale"] = 0.84 * img_width / 1280
            self.display_scales["countdown_font_scale"] = 10 * img_width / 1280
```
If this is viable, you could consider including the different text thicknesses in `self.display_scales` and rename this dictionary as well. Let me know what you think?